Marlin/src/module/stepper.cpp

```diff
@@ -2460,7 +2460,20 @@ hal_timer_t Stepper::block_phase_isr() {
 
         #if ENABLED(LIN_ADVANCE)
           if (la_active) {
-            const uint32_t la_step_rate = la_advance_steps < current_block->max_adv_steps ? current_block->la_advance_rate : 0;
+            #if ENABLED(LA_ZERO_SLOWDOWN)
+              const uint32_t step_events_left_until_ramp_down = accelerate_before - step_events_completed;
+              const uint32_t max_e_accel =  STEP_MULTIPLY(interval, current_block->max_e_acc); 
+              // TODO: respect max e accel by also considering the acceleration from the step_rate delta between two timer steps
```

Comment:
The TODO is about using `max_e_accel - step_rate_acceleration` on ramp up, and `max_e_accel + step_rate_acceleration` on ramp down, since `step_rate_acceleration` is also seen by the extruder

Reply:
Something like 
```cpp
const uint32_t acc_delta_v = STEP_MULTIPLY(interval, current_block->acceleration_rate);
const uint32_t max_delta_v_ramp_up = max_delta_v - acc_delta_v;
const uint32_t max_delta_v_ramp_down = max_delta_v + acc_delta_v;
const bool is_ramping_up = step_events_left_in_phase * max_delta_v_ramp_down >= current_block->current_la_step_rate;

if (is_ramping_up){
  current_block->current_la_step_rate += _MIN(max_delta_v_ramp_up, current_block->la_advance_rate - current_block->current_la_step_rate);
} else {
  current_block->current_la_step_rate -= _MIN(max_delta_v_ramp_down, current_block->current_la_step_rate);
}
```

Reply:
This could have been covered in planner.cpp I think when initially calculating `current_block->max_e_acc`.